grapple/types/redirects.py

```diff
@@ -1,37 +1,61 @@
+import json
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
 
 from .pages import get_page_interface
 
 
-class RedirectType(graphene.ObjectType):
+class RedirectObjectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
```

Comment:
As discussed just now, let's simplify this.

Let's say you have a `Redirect` for two sites. Currently, you would return a single object with multiple values for `site` and `old_url`:

```
Redirect(
    old_url=["http://xx.com/hello", "http://yy.com/hello"]
    site=[x,y]
    ...
)
```

But this is confusing because the field names suggest singular values. Also, API consumers need to loop through `Redirect`s and then loop through each redirect's sites/URLs to set up redirect handling within their codebase, which is arduous.

To simplify this, can you return a separate object for each site/url? For example, the above response would become:

```
Redirect(
    old_url="http://xx.com/hello"
    site=x
    ...
)
Redirect(
    old_url="http://yy.com/hello"
    site=y
    ...
)
```

Reply:
Implemented this in 75707b2fcdd68d10feccadbeca9a62755fe8bade

When returning a list of redirects, for redirects with `site=None` we now create a copy for each existing site and then remove the one with no site from the list.